keps/sig-node/2400-node-swap/README.md

```diff
@@ -0,0 +1,727 @@
+|MemorySwappiness|This configuration sets how aggressively the kernel will swap memory pages. By default, the host kernel can swap out a percentage of anonymous pages used by a container. Users can set value between 0 and 100, to tune this percentage.|Unset, which will use host value|SupportNodeMemorySwap|
+
+#### MemorySwapLimit details
+
+MemorySwapLimit configuration is a kubelet flag that only takes effect on a
+container that has a memory limit set, either explicitly from
+[PodSpec]([https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#requests-and-limits](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#requests-and-limits)
+) or implicitly from [Resource
+Quota]([https://kubernetes.io/docs/concepts/policy/resource-quotas/](https://kubernetes.io/docs/concepts/policy/resource-quotas/)
+).
+
+For container with memory limit set, MemorySwapLimit setting will have the
+following effects, [similar to
+docker](https://docs.docker.com/config/containers/resource_constraints/#--memory-swap-details)
+
+* If MemorySwapLimit is set to a positive integer,
+  * If the memory limit of the container is greater or equal to
+    MemorySwapLimit, then no swap is allowed, the container does not have
+    access to swap.
+  * If the memory limit of the container is less than MemorySwapLimit, then
+    MemorySwapLimit represents the total amount of memory and swap that can be
+    used. For example, for a container with memory limit set to 300m, and
+    `MemorySwapLimit` set to 1g, the container can use 300m of memory and 700m (1g
+    - 300m) swap.
+* If MemorySwapLimit is set to 0, for containers with memory limit is set, the
+  container can use as much swap as the Memory limit setting, if the host
+  container has swap memory configured. For instance, if  a container requests
+  memory="300m" and MemorySwapLimit is not set, the container can use 600m in
+  total of memory and swap.
+* If MemorySwapLimit is explicitly set to -1, the container is allowed to use
+  unlimited swap, up to the amount available on the host system.
+* If MemorySwapLimit is explicitly set to -2,  the container does not have
+  access to swap. This value effectively prevents a container from using swap.
+
+In summary, for users experimenting with this feature
+
+|MemorySwapLimit|container memory limit (explicit or implicit)|Expected Behavior|Comment|
+|--- |--- |--- |--- |
+|Any|not set|N/A|Same as docker|
+|-2|N|no swap allowed, this is the default value||
+|-1|N|unlimited swap|Same as docker|
+|0|N|container can use up to N swap (ie: 2N memory+swap)|Same as docker|
+|X where X > 0|N where N < X|container can use up to X-N swap (ie: 2N memory+swap)|Same as docker|
+|X where X > 0|N where N >= X|no swap allowed (ie: N memory only)|Same as docker|
+
+#### MemorySwappiness details
```

Comment:
As I commented above, I don't think we should include `MemorySwappiness`.

Reply:
Ack. Let's use the first comment as a basis for discussion to avoid split context.